thorlcr/slave/slwatchdog.cpp

```diff
@@ -132,6 +132,29 @@ class CGraphProgressHandlerBase : public CSimpleInterface, implements ISlaveWatc
             activeGraphs.zap(graph);
         }
     }
+    virtual void debugRequest(CMessageBuffer &msg, const char *request) const
+    {
+        Owned<IPTree> req = createPTreeFromXMLString(request);
+        activity_id actId = req->getPropInt("@source", 0);
+        unsigned edgeIdx = req->getPropInt("@edge", 0);
+        CriticalBlock b(crit);
+        ForEachItemIn(g, activeGraphs) // NB: 1 for each slavesPerProcess
+        {
+            // MORE - this code should be in CGraphBase
+            CGraphBase &graph = activeGraphs.item(g);
+            CGraphElementBase *element = graph.queryElement(actId);
```

Comment:
I think the above 'MORE - this code should be in CGraphBase' is still true, would be better if moved into CSlaveGraph